aten/src/ATen/functorch/BatchRulesDecompositions.cpp

```diff
@@ -308,6 +308,15 @@ TORCH_LIBRARY_IMPL(aten, FuncTorchBatchedDecomposition, m) {
   OP_DECOMPOSE(swapdims_);
   OP_DECOMPOSE(swapaxes_);
   OP_DECOMPOSE(unfold_copy);
+  m.impl("upsample_bilinear2d.vec", native::upsample_bilinear2d);
+  m.impl("upsample_bicubic2d.vec", native::upsample_bicubic2d);
+  m.impl("_upsample_bilinear2d_aa.vec", native::_upsample_bilinear2d_aa);
+  m.impl("_upsample_bicubic2d_aa.vec", native::_upsample_bicubic2d_aa);
+  m.impl("upsample_linear1d.vec", native::upsample_linear1d);
+  m.impl("upsample_nearest1d.vec", native::upsample_nearest1d);
+  m.impl("upsample_nearest2d.vec", native::upsample_nearest2d);
+  m.impl("upsample_nearest3d.vec", native::upsample_nearest3d);
+  m.impl("upsample_trilinear3d.vec", native::upsample_trilinear3d);
```

Comment:
Why aren't these OP_DECOMPOSE2 ?

Reply:
It is possible that we can use `OP_DECOMPOSE2` but it would require a static cast of the function pointer:
```c++
#define OP_DECOMPOSE2(op, overload)  m.impl(#op"."#overload, static_cast<decltype(&ATEN_FN2(op, overload))>(native::op));
```
and I would certainly need to introduce some `*_symint` methods and adapt the symint structures into original functions. Also we may add this to `native_functions.yaml`:
```
- func: upsample_bilinear2d.vec(Tensor input, SymInt[]? output_size, bool align_corners, float[]? scale_factors) -> Tensor
  python_module: nn
  autogen: upsample_bilinear2d.vec_out
  tags: core
  dispatch:
    CompositeImplicitAutograd: upsample_bilinear2d_symint

```

I'm not sure what would be the best way to do that. I found doing this way we do not introduce more (maybe redundant) code. Let me know if we should introduce `*_symint` variants and use `OP_DECOMPOSE2`

Reply:
Let's just do the `m.impl`, it's much easier and it works

Reply:
Can you leave a quick comment about why these are m.impl into the code?